src/v3/plugins/github/createGraph.js

```diff
@@ -0,0 +1,167 @@
+  reviewCommentUrlToId,
+} from "./urlIdParse";
+
+export function createGraph(data: GithubResponseJSON): Graph {
+  const creator = new GraphCreator();
+  creator.addData(data);
+  return creator.graph;
+}
+
+class GraphCreator {
+  graph: Graph;
+
+  constructor() {
+    this.graph = new Graph();
+  }
+
+  addNode(addr: StructuredAddress) {
+    this.graph.addNode(toRaw(addr));
+  }
+
+  addData(data: GithubResponseJSON) {
+    this.addRepository(data.repository);
+  }
+
+  addRepository(repoJSON: RepositoryJSON) {
+    const repo: RepoAddress = {
+      type: "REPO",
+      owner: repoJSON.owner.login,
+      name: repoJSON.name,
+    };
+    this.addNode(repo);
+    repoJSON.issues.nodes.forEach((issue) => this.addIssue(repo, issue));
+    repoJSON.pulls.nodes.forEach((pull) => this.addPull(repo, pull));
+  }
+
+  addIssue(repo: RepoAddress, issueJSON: IssueJSON) {
+    const issue: IssueAddress = {
+      type: "ISSUE",
+      repo,
+      number: String(issueJSON.number),
+    };
+    this.addNode(issue);
+    this.addAuthors(issue, issueJSON.author);
+    this.addHasParent(issue, repo);
+    issueJSON.comments.nodes.forEach((comment) =>
+      this.addComment(issue, comment)
+    );
+  }
+
+  addPull(repo: RepoAddress, pullJSON: PullJSON) {
+    const pull: PullAddress = {
+      type: "PULL",
+      repo,
+      number: String(pullJSON.number),
+    };
+    this.addNode(pull);
+    this.addAuthors(pull, pullJSON.author);
+    this.addHasParent(pull, repo);
+    pullJSON.comments.nodes.forEach((comment) =>
+      this.addComment(pull, comment)
+    );
+    pullJSON.reviews.nodes.forEach((review) => this.addReview(pull, review));
+    if (pullJSON.mergeCommit != null) {
+      const commitHash = pullJSON.mergeCommit.oid;
+      // TODO: Replace with an API from the v3 Git plugin
+      const commit = NodeAddress.fromParts([
+        "sourcecred",
+        "git",
+        "commit",
+        commitHash,
+      ]);
+      this.graph.addNode(commit);
+      this.graph.addEdge(createEdge.mergedAs(pull, commit));
+    }
+  }
+
+  addReview(pull: PullAddress, reviewJSON: ReviewJSON) {
+    const id = reviewUrlToId(reviewJSON.url);
+    const review = {
+      type: "REVIEW",
+      pull,
+      id,
+    };
+    this.addNode(review);
+    reviewJSON.comments.nodes.forEach((comment) =>
+      this.addComment(review, comment)
+    );
+    this.addAuthors(review, reviewJSON.author);
+    this.addHasParent(review, pull);
+  }
+
+  addComment(
+    parent: IssueAddress | PullAddress | ReviewAddress,
+    commentJSON: CommentJSON | ReviewCommentJSON
+  ) {
+    const id = (function() {
+      switch (parent.type) {
+        case "ISSUE":
+          return issueCommentUrlToId(commentJSON.url);
+        case "PULL":
+          return pullCommentUrlToId(commentJSON.url);
+        case "REVIEW":
+          return reviewCommentUrlToId(commentJSON.url);
+        default:
+          throw new Error(`Unexpected comment parent type: ${parent.type}`);
```

Comment:
This can be `(parent.type: empty)`, right?

Reply:
Yep, added.